CIP-????/README.md

```diff
@@ -0,0 +1,334 @@
+
+The liabilities paper also proposes an on-chain intent-processing system for Babel-fee transactions. 
+As previously discussed, this proposal is only focused on intent settlement, not intent processing.
+
+#### Negative outputs
+
+This proposal is functionally identical to a variant where instead of having unresolved inputs, we have negative outputs. 
+Negative outputs differ from liabilities in that they are a separate type of output which contains values with only negative quantities.
+
+The advantages of the negative outputs view are:
+
+- Rather than having a separate set of UTXIs, we can consider unresolved negative inputs as a kind of UTXO
+- Unresolved inputs behave similarly to outputs in many ways, e.g. having credentials to control their usage
+
+The disadvantages of the negative outputs view are:
+
+- We need to introduce negative quantities, all ledger rules and scripts need to be aware of these. In contrast, unresolved inputs look more like normal inputs during validation of the transaction itself.
+- The linearizability condition looks backwards, since we would require negative outputs to be ordered after the inputs which “spend” them. Similarly ,the argument that we can view the linearized order as corresponding to a “normal” series of transactions is more subtle, since we would also need to translate negative outputs into inputs for that to be true.
+
+The authors’ position is that: 
+
+- Unresolved inputs are conceptually clearer
+- Both UTXOs and UTXIs are “unresolved”, and hence need access control. This is a fine explanation of their similarities, and doesn’t mean that UTXIs are “really” outputs.
+
+#### Westberg’s “Smart Transactions”
+
+The design from https://github.com/input-output-hk/Developer-Experience-working-group/issues/47 has a lot of similarities to ours, in particular it follows the approach where transactions have partially unspecified inputs and outputs. 
+However, it goes much further and proposes conditions on how those can be satisfied, as well as some kind of computed conditions for updating datums. 
+It also proposes that intent-processing be done by the node.
+
+We think that we can eventually move in the direction of adding more of the intent-settlement features from the proposal (e.g. conditions on values in inputs), but we should do so incrementally: even the machinery we need to support the more limited intents in this proposal is quite substantial.
+
+A major point of disagreement is about intent processing. 
+As previously discussed, this proposal only focuses on intent settlement, not intent processing.
+
+#### A better design
+
+It may be that the best way to implement intents on Cardano is different from the approach we are taking (transactions which leave more details unspecified). 
+In that case such a design might entirely obsolete this one, making it an expensive and pointless burden to maintain in future.
```

Comment:
For sure! But one thing we can consider to try to make the current design more compatible with more expressive intents is replace the `IsBackwards` field of `TxOutZone` with some other more expressive metadata that specifies what sort of intent this is. For example, if we want an unspecified datum, `MissingDatum` could be the tag on an output.